src/engine/core/lua_wrapper.h

```diff
@@ -288,9 +288,9 @@ template <int N> struct FunctionCaller
 											 lua_State* L,
 											 Args... args)
 	{
-		typedef std::tuple_element<sizeof...(ArgsF)-N,
-								   std::tuple<ArgsF...>>::type T;
-		typedef std::remove_cv<std::remove_reference<T>::type>::type RealT;
+		typedef typename std::tuple_element<sizeof...(ArgsF)-N,
```

Comment:
By standard, you have to provide `typename` for type members in template class. Visual Studio does a good job at figuring it out. clang and gcc are not so good here.